api/src/commonMain/kotlin/dev/kord/cache/api/observables/Relation.kt

```diff
@@ -0,0 +1,35 @@
+package dev.kord.cache.api.observables
+
+/**
+ * A typealias for a function that determines the relationship between two entities of type `T` and `R`
+ * in a uni-directional link.
+ * The `value` parameter represents the entity of type `T` and the `friend` parameter represents the entity of type `R`.
+ * The function should return `true` if the two entities are related, or `false` otherwise.
+ */
+public typealias RelationHandler<T, R> = (value: T, friend: R) -> Boolean
+
+/**
+ * A `Relation` is a uni-directional link between two entities of type `T` and `R`.
+ * A `Relation` is defined by a set of `RelationHandler`s which determine the relationship
+ * between two entities.
+ *
+ * @param T the type of the first entity in the relation.
+ */
+public interface Relation<T: Any> {
```

Comment:
does this need to be a public interface now? it's no longer exposed by the other public interfaces so it could now just be an implementation detail

Reply:
It may be provided to customize the deletion process.
I've already added it to the implementation
What do you suggest we do with this one?